var/spack/repos/builtin/packages/hdf5/package.py

```diff
@@ -41,6 +41,7 @@ class Hdf5(CMakePackage):
 
     # Odd versions are considered experimental releases
     # Even versions are maintenance versions
+    version("1.14.2", tag="hdf5-1_14_2", preferred=True)
```

Comment:
Why are we using tags instead of tarballs?

Reply:
We wanted to check the CI testing before we were ready to publish the tarballs.  I'll update the PR with tarball and checksum in the next hour.

Reply:
downloading source tar.gz file now with sha256 checksum 1c342e634008284a8c2794c8e7608e2eaf26d01d445fb3dfd7f33cb2fb51ac53:
==> Installing hdf5-1.14.2-7isfblh67bj2xh7hr7wni43fknqzwrp2 [14/14]
==> No binary for hdf5-1.14.2-7isfblh67bj2xh7hr7wni43fknqzwrp2 found: installing from source
==> Fetching https://support.hdfgroup.org/ftp/HDF5/releases/hdf5-1.14/hdf5-1.14.2/src/hdf5-1.14.2.tar.gz